### PR TITLE
Fix DynamicIdentityFileCreds being incompatible with L7 Loadbalancers

### LINKED ISSUE #36229: Slack plugin fails to connect with "refresh_identity = true" to a cluster behind ALB
Expected behavior:
Slack plugin will work with `refresh_identity = true` on clusters where the proxies are behind L7 loadbalancers like AWS ALB with ACM certificates.

Current behavior:
When `refresh_identity = true` is set, the slack plugin fails to connect.

Workaround:
use v14.1.0 or disable refresh_identity = true and use long lived credentials

Bug details:
- Teleport version
anything >14.1.0 plugin version, suspected to be related to changes in 14.1.3

- Recreation steps
deploy slack plugin with `refresh_identity = true` and attempt to connect to a cluster behind AWS ALB with ACM

- Debug logs
~~~
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana systemd[1]: Started Teleport Slack Plugin.
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: INFO   Starting Teleport Access Slack Plugin 14.2.1: slack/main.go:91
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: WARN   Configuration setting `auth_server` is deprecated, consider to change it to `addr` lib/config.go:150
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:21 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:22 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:22 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:22 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:22 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:22 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:23 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:23 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:23 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:23 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:24 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:25 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:25 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:25 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:26 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:26 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:27 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:27 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:27 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:27 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:28 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:29 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:29 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade required for "test-cluster1.plainsofconquest.com:443": true. client/alpn_conn_upgrade.go:95
Jan 03 15:49:29 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   ALPN connection upgrade for test-cluster1.plainsofconquest.com:443. client/alpn_conn_upgrade.go:187
Jan 03 15:49:30 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:30 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: DEBU   Attempting GET test-cluster1.plainsofconquest.com:443/webapi/find webclient/webclient.go:129
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [1]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: *errors.errorString context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, use of closed network connection&#34;
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:485 github.com/gravitational/teleport/api/client.(*Client).dialGRPC
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:409 github.com/gravitational/teleport/api/client.tlsRoutingConnect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:234 github.com/gravitational/teleport/api/client.connect.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         failed to connect to addr test-cluster1.plainsofconquest.com:443 with TLS Routing dialer
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:                 context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, use of closed network connection&#34;] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [2]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: *errors.errorString context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: tls: failed to verify certificate: x509: certificate signed by unknown authority&#34;
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:485 github.com/gravitational/teleport/api/client.(*Client).dialGRPC
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:423 github.com/gravitational/teleport/api/client.tlsRoutingWithConnUpgradeConnect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:234 github.com/gravitational/teleport/api/client.connect.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         failed to connect to addr test-cluster1.plainsofconquest.com:443 with TLS Routing with ALPN connection upgrade dialer
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:                 context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: tls: failed to verify certificate: x509: certificate signed by unknown authority&#34;] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [3]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: *errors.errorString context deadline exceeded: connection error: desc = &#34;transport: authentication handshake failed: x509: certificate is valid for test-cluster1.plainsofconquest.com, not teleport.cluster.local&#34;
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:485 github.com/gravitational/teleport/api/client.(*Client).dialGRPC
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:370 github.com/gravitational/teleport/api/client.authConnect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:234 github.com/gravitational/teleport/api/client.connect.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         failed to connect to addr test-cluster1.plainsofconquest.com:443 as an auth server
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:                 context deadline exceeded: connection error: desc = &#34;transport: authentication handshake failed: x509: certificate is valid for test-cluster1.plainsofconquest.com, not teleport.cluster.local&#34;] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [4]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: *errors.errorString context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, close tcp 172.31.94.174:34592-&gt;54.80.66.60:443: use of closed network connection&#34;
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:485 github.com/gravitational/teleport/api/client.(*Client).dialGRPC
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:396 github.com/gravitational/teleport/api/client.proxyConnect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:234 github.com/gravitational/teleport/api/client.connect.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         failed to connect to addr test-cluster1.plainsofconquest.com:443 as a web proxy
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:                 context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, close tcp 172.31.94.174:34592-&gt;54.80.66.60:443: use of closed network connection&#34;] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [5]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: *errors.errorString context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, close tcp 172.31.94.174:36994-&gt;34.234.128.129:443: use of closed network connection&#34;
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:485 github.com/gravitational/teleport/api/client.(*Client).dialGRPC
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:383 github.com/gravitational/teleport/api/client.tunnelConnect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:234 github.com/gravitational/teleport/api/client.connect.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         failed to connect to addr test-cluster1.plainsofconquest.com:443 as a reverse tunnel proxy
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:                 context deadline exceeded: connection error: desc = &#34;transport: Error while dialing: failed to dial: ssh: handshake failed: ssh: overflow reading version string, close tcp 172.31.94.174:36994-&gt;34.234.128.129:443: use of closed network connection&#34;] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERRO   Terminating with fatal error [6]... error:[
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: ERROR REPORT:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Original Error: context.deadlineExceededError context deadline exceeded
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: Stack Trace:
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:344 github.com/gravitational/teleport/api/client.connect
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport/api@v0.0.0-20231201005414-e6737fe00510/client/client.go:144 github.com/gravitational/teleport/api/client.New
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/lib/config.go:182 github.com/gravitational/teleport/integrations/lib.TeleportConfig.NewClient
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/access/common/config.go:67 github.com/gravitational/teleport/integrations/access/common.GetTeleportClient
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/access/common/config.go:77 github.com/gravitational/teleport/integrations/access/common.BaseConfig.GetTeleportClient
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/access/common/app.go:103 github.com/gravitational/teleport/integrations/access/common.(*BaseApp).initTeleport
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/access/common/app.go:181 github.com/gravitational/teleport/integrations/access/common.(*BaseApp).init
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/access/common/app.go:133 github.com/gravitational/teleport/integrations/access/common.(*BaseApp).run
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/lib/process.go:230 main.run.NewSlackApp.NewApp.NewServiceJob.func1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/lib/process.go:251 github.com/gravitational/teleport/integrations/lib.(*serviceJob).DoJob
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /go/pkg/mod/github.com/gravitational/teleport@v0.0.0-20231201005414-e6737fe00510/integrations/lib/process.go:97 github.com/gravitational/teleport/integrations/lib.NewProcess.func2.1
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]:         /usr/local/go/src/runtime/asm_amd64.s:1650 runtime.goexit
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana teleport-slack[21763]: User Message: context deadline exceeded] lib/bail.go:28
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana systemd[1]: teleport-slack.service: Main process exited, code=exited, status=1/FAILURE
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana systemd[1]: teleport-slack.service: Failed with result 'exit-code'.
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana systemd[1]: teleport-slack.service: Service RestartSec=100ms expired, scheduling restart.
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana systemd[1]: teleport-slack.service: Scheduled restart job, restart counter is at 27.
Jan 03 15:49:31 Test-Cluster-Node2-US-East-Grafana systemd[1]: Stopped Teleport Slack Plugin.
~~~

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/36229

changelog: Fixes `refresh_identity = true` preventing Access Plugins connecting to Teleport using TLS routing with a L7 LB.

### CODE REVIEW

File: api/client/contextdialer.go [hunk 1]
```diff
@@ -364,6 +364,9 @@ func newTLSRoutingWithConnUpgradeDialer(ssh ssh.ClientConfig, params connectPara
 			},
 			ALPNConnUpgradeRequired: IsALPNConnUpgradeRequired(ctx, params.addr, insecure),
 			GetClusterCAs: func(_ context.Context) (*x509.CertPool, error) {
+				// TODO(noah): there's potentially a bug here when multiple
+				// credentials have been configured as the CAs are always
+				// retrieved from the first credential.
```
Comment (greedy52) [inline api/client/contextdialer.go:369]: hmm.. good catch. Probably can just use `params.tlsConfig.RootCAs`?

**Review by greedy52 — APPROVED** [review]

File: api/client/contextdialer.go (hunk 1 above)
Comment (strideynet) [reply api/client/contextdialer.go:369]: That looks like a good fix - I'll pop that in and test it :)

Comment (strideynet) [reply api/client/contextdialer.go]: Works nicely 🥳  [b22c6e5](https://github.com/gravitational/teleport/pull/36411/commits/b22c6e585fc857ff0a9f56dcf74af2dbfed71100)

**Comment by strideynet** [comment]
Tested locally with 

```go
package main

import (
	"context"
	glog "log"
	"os"
	"os/signal"

	"github.com/sirupsen/logrus"

	"github.com/gravitational/teleport/api/client"
	apidefaults "github.com/gravitational/teleport/api/defaults"
	"github.com/gravitational/trace"
)

func main() {
	err := run()
	glog.Fatal(err)
}

func run() error {
	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
	defer cancel()

	log := logrus.StandardLogger()
	logrus.SetLevel(logrus.DebugLevel)
	log.SetLevel(logrus.DebugLevel)

	creds, err := client.NewDynamicIdentityFileCreds("./admin_identity")
	if err != nil {
		return trace.Wrap(err, "loading creds")
	}
	teleportClient, err := client.New(ctx, client.Config{
		Addrs:       []string{"leaf.tele.ottr.sh:443"},
		Credentials: []client.Credentials{creds},
	})
	if err != nil {
		return trace.Wrap(err, "connecting")
	}
	defer teleportClient.Close()

	nodes, err := teleportClient.GetNodes(ctx, apidefaults.Namespace)
	if err != nil {
		return trace.Wrap(err, "get nodes")
	}
	log.Infof("Nodes: %d", len(nodes))
	return nil
}
```

File: api/client/contextdialer.go (hunk 1 above)
Comment (greedy52) [reply api/client/contextdialer.go]: thanks! 🙏

File: api/client/credentials.go [hunk 2]
```diff
@@ -473,7 +475,10 @@ func (d *DynamicIdentityFileCreds) TLSConfig() (*tls.Config, error) {
 		},
 
 		// VerifyConnection is used instead of the static RootCAs field.
-		RootCAs: nil,
+		// However, there's some client code which relies on the static RootCAs
+		// field. So we set it to a copy of the current root CAs pool to support
+		// those - e.g ALPNDialerConfig.GetClusterCAs
+		RootCAs: d.tlsRootCAs.Clone(),
```
Comment (codingllama) [inline api/client/credentials.go:481]: Is there a practical way we could unit test this?

**Review by codingllama — APPROVED** [review]

File: api/client/credentials.go (hunk 2 above)
Comment (strideynet) [reply api/client/credentials.go:481]: I can add a rough unit test for this I believe - I'll do that shortly

Comment (strideynet) [reply api/client/credentials.go:481]: 213597e082

File: api/client/credentials_test.go [hunk 3]
```diff
@@ -416,6 +416,13 @@ func TestDynamicIdentityFileCreds(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, wantTLSCert, *gotTLSCert)
 
+	tlsCACertPEM, _ := pem.Decode(tlsCACert)
+	tlsCACertDER, err := x509.ParseCertificate(tlsCACertPEM.Bytes)
+	require.NoError(t, err)
+	wantCertPool := x509.NewCertPool()
+	wantCertPool.AddCert(tlsCACertDER)
+	require.True(t, wantCertPool.Equal(tlsConfig.RootCAs))
```
Comment (codingllama) [inline api/client/credentials_test.go:424]:
```suggestion
	require.True(t, wantCertPool.Equal(tlsConfig.RootCAs), "tlsconfig.RootCAs mismatch")
```

**Review by codingllama — APPROVED** [review]

**Pull request merged**